### PR TITLE
Closes #96

### LINKED ISSUE #96: Create a dataset loader for SETH-corpus
- Task: NER
- License: Apache License, Version 2.0
- Format: brat
- Language: English
- Citation:
@Article{SETH2016,
  Title= {SETH detects and normalizes genetic variants in text.},
  Author= {Thomas, Philippe and Rockt{\"{a}}schel, Tim and Hakenberg, J{\"{o}}rg and Lichtblau, Yvonne and Leser, Ulf},
  Journal= {Bioinformatics},
  Year= {2016},
  Month= {Jun},
  Doi= {10.1093/bioinformatics/btw234},  
  Language = {eng},
  Medline-pst = {aheadofprint},
  Pmid = {27256315},
  Url = {http://dx.doi.org/10.1093/bioinformatics/btw234}
}
- Source: https://github.com/rockt/SETH/tree/master/resources/SETH-corpus

### PR DESCRIPTION
**Pull request by vumichien**
Closes #96 
### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.

### CODE REVIEW

File: biodatasets/seth_corpus/seth_corpus.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                example["id"] = str(guid)
+                example["text"] = text
+                yield guid, example
+        elif self.config.schema == "bigbio_kb":
+            with open(filepath/corpus_file, encoding='utf-8') as f:
+                contents = f.readlines()
+            for guid, content in enumerate(contents):
+                file_name, text = content.split("\t")
+                example = parsing.parse_brat_file(filepath / "annotations" / f"{file_name}.ann")
+                print(example)
```
Comment (sg-wbi) [inline biodatasets/seth_corpus/seth_corpus.py:216]: could you please remove the print statetments?

File: utils/parsing.py [hunk 2]
```diff
@@ -265,6 +265,8 @@ def brat_parse_to_bigbio_kb(brat_parse: Dict, entity_types: Iterable[str]) -> Di
     id_to_event_trigger = {}
     for ann in brat_parse["text_bound_annotations"]:
         if ann["type"] in entity_types:
+            if len(brat_parse["events"]) > 0:
```
Comment (sg-wbi) [inline utils/parsing.py:268]: I am not sure if this is an update from the repo or you modified it. 
In the latter case, if you feel there is a bug in `parsing.py` I suggest you make a separate PR for that.

**Review by sg-wbi — COMMENTED** [review]
Thank you for your contribution! The dataset loader seems to work fine.
Could you please address the comments I made? Thanks again!

File: utils/parsing.py (hunk 2 above)
Comment (vumichien) [reply utils/parsing.py:268]: I see. I will remove this line and copy the modified function into `seth_corpus.py`. It's the better way to avoid conflict with other files.

File: biodatasets/seth_corpus/seth_corpus.py (hunk 1 above)
Comment (vumichien) [reply biodatasets/seth_corpus/seth_corpus.py]: sorry, I forgot to remove them

File: biodatasets/seth_corpus/seth_corpus.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+_HOMEPAGE = "https://github.com/rockt/SETH"
+
+_LICENSE = "Apache License 2.0"
+
+_URLS = {
+    "source": "https://github.com/rockt/SETH/archive/refs/heads/master.zip",
+    "bigbio_kb": "https://github.com/rockt/SETH/archive/refs/heads/master.zip",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
```
Comment (leonweber) [inline biodatasets/seth_corpus/seth_corpus.py:60]: `_SUPPORTED_TASKS` should include `Tasks.RELATION_EXTRACTION`

**Review by leonweber — CHANGES_REQUESTED** [review]
supported tasks is missing relation extraction. Otherwise it looks good to go for me.

File: biodatasets/seth_corpus/seth_corpus.py (hunk 3 above)
Comment (ChienVM) [reply biodatasets/seth_corpus/seth_corpus.py]: This task was added to the script

Comment (vumichien) [reply biodatasets/seth_corpus/seth_corpus.py]: This task was added in the script.

**Review by leonweber — APPROVED** [review]
Looks good to me now. Thanks a lot for your contribution!
